mesonbuild/build.py

```diff
@@ -385,6 +386,10 @@ def process_kwargs(self, kwargs, environment):
         if not isinstance(valalist, list):
             valalist = [valalist]
         self.add_compiler_args('vala', valalist)
+        dlist = kwargs.get('d_args', [])
+        if not isinstance(dlist, list):
+            dlist = [dlist]
+        self.add_compiler_args('d', dlist)
```

Comment:
Nice! In that case, this should probably be used for all the other language flags as well...